### DISCUSSION TITLE
problems in shadows, in gradient mesh

### DISCUSSION THREAD

**Discussion by renato3xl** [bnpr/Malt#196] [Q&A] [answered]
Hi! i started a new scene and after the animation i realized i can't see any projected shadow .
I use the override option and the gradient mesh shader, that used to give me a white base and black self and projected shadows, but now there is no way i can see the projected ones. i've tried other shaders and non has shadows but this one worked for me in the past.

Any idea on what can be?

Thanks a lot!

**Comment by HSDSZ** [comment]
drag the white/black boudary  to any value larger than 0.5

**Reply by pragma37** [reply]
Yes, this is most likely your problem.
![imagen](https://user-images.githubusercontent.com/36610999/145679320-cafd3cbd-ea65-433a-9d76-cbe60306fb46.png)
![imagen](https://user-images.githubusercontent.com/36610999/145679336-a2c67bba-2e21-416b-bbb0-60bf4eebd8b6.png)

If it's not, then upload a file where I can reproduce the issue, so I can take a look at it.

**Reply by renato3xl** [reply]
i`ve tried and no shadows :S 
![image](https://user-images.githubusercontent.com/94867187/145679785-9983fe05-175f-4a17-94c1-c4289c35e6c5.png)

**Reply by renato3xl** [reply]
Ok... i have created a cube and a plane and there i have shadows... 
any idea why there aren't in the ships?¿

![image](https://user-images.githubusercontent.com/94867187/145679887-90a940e6-5bca-42bc-8d97-e7b973bcfd08.png)

**Reply by pragma37** [reply]
I don't think there are any mesh setting that could affect shadowmaps.
Could be the shadowmaps settings like in your previous question?
https://github.com/bnpr/Malt/discussions/172#discussioncomment-1688078

If you are ok with sending me the file I can take a look into it.

**Reply by renato3xl** [reply]
sure, can you give me an email? the file i think is quite big

**Reply by pragma37** [reply]
pragma37@gmail.com
You can also dm me on [twitter](https://twitter.com/pragma37).

**Reply by eballeste** [reply]
If you guys figure this out, could you share what happened here? I somehow was experiencing this same issue and was only able to get shadows to work by starting the scene all over again.

**Reply by pragma37** [reply]
@eballeste See https://github.com/bnpr/Malt/discussions/196#discussioncomment-1799148

If that's not your problem, open a new thread and upload a minimal reproducible example file so I can look into it. 👍

**Comment by pragma37** [comment, marked as answer]
The problem was that the Max Distance value in the Sun ShadowMaps settings was too low.
If you find the same issue increase the value until it covers your whole scene and increase shadow cascades/resolution accordingly.

![imagen](https://user-images.githubusercontent.com/36610999/145841394-8a18ea46-ebf9-4ae6-b062-387edd5aaf72.png)
